engine/order_manager.go

```diff
@@ -284,6 +284,49 @@ func (m *OrderManager) validate(newOrder *order.Submit) error {
 	return nil
 }
 
+// Modify depends on the order.Modify.ID and order.Modify.Exchange fields to uniquely
+// identify an order to modify.
+func (m *OrderManager) Modify(mod *order.Modify) (*order.ModifyResponse, error) {
+	if m == nil {
+		return nil, fmt.Errorf("order manager %w", ErrNilSubsystem)
+	}
+	if atomic.LoadInt32(&m.started) == 0 {
+		return nil, fmt.Errorf("order manager %w", ErrSubSystemNotStarted)
+	}
+
+	exch := m.orderStore.exchangeManager.GetExchangeByName(mod.Exchange)
+	if exch == nil {
+		return nil, ErrExchangeNotFound
+	}
+
+	id, err := exch.ModifyOrder(mod)
+	if err != nil {
+		return nil, err
+	}
+
+	// We now have to reflect this change in GCT's internal state of managed orders.
+
+	// TODO: Current implementation is here for illustrative purposes.  It introduces
+	// race conditions.
+	det, err := m.orderStore.getByExchangeAndID(mod.Exchange, mod.ID)
```

Comment:
We don't have an update procedure just yet that doesn't cause a race condition with order.Modify properties. 
You can also see in [websocketroutine_manager.go](https://github.com/thrasher-corp/gocryptotrader/blob/master/engine/websocketroutine_manager.go#L234) the POC implementation, that this is not ideal and this needs to change.

We can continue implementing features like this but a broad update on the order management system will need to be implemented and will be one of our priorities moving forward.